### PR TITLE
Set PL_BACKEND in L-GPU docker build and misc improvements

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
The Lightning-GPU Docker image has not installed the wheel correctly as the wheel-building command was missing the environment variable `PL_BACKEND=lightning_gpu`. 

**Description of the Change:**
Set the environment variable `PL_BACKEND=lightning_gpu` in the `build-wheel-lightning-gpu` Docker-build stage.
Enhance Docker-related `make` commands.
Enable Dockerfile level defaults for build arguments.
Set the default Lightning and PennyLane versions to `master`.

**Benefits:**
L-GPU installs the correct wheels.
No need to update hardcoded tags/versions since `master` is moving.

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-68127]

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -143,15 +152,16 @@ RUN pip install --no-cache-dir git+https://github.com/PennyLaneAI/pennylane.git@
 # Download and build Lightning-GPU release
 FROM base-build-cuda AS build-wheel-lightning-gpu
 WORKDIR /opt/pennylane-lightning
+ENV PL_BACKEND=lightning_gpu
 RUN pip install --no-cache-dir wheel custatevec-cu12
 RUN pip uninstall -y pennylane-lightning
-RUN CUQUANTUM_SDK=$(python -c "import site; print( f'{site.getsitepackages()[0]}/cuquantum/lib')") PL_BACKEND=lightning_gpu python setup.py build_ext
+RUN CUQUANTUM_SDK=$(python -c "import site; print( f'{site.getsitepackages()[0]}/cuquantum/lib')") python setup.py build_ext
```
Comment (multiphaseCFD) [inline docker/Dockerfile:158]: Can we set CUQUANTUM_SDK as ENV as well?

File: docker/Dockerfile [hunk 2]
```diff
@@ -12,18 +12,27 @@
...
 ARG ROCM_INSTALLER=https://repo.radeon.com/amdgpu-install/5.7/ubuntu/jammy/amdgpu-install_5.7.50700-1_all.deb
+
+# Create basic runtime environment base on Ubuntu 22.04 (jammy)
+# Create and activate runtime virtual environment
+FROM ubuntu:jammy AS base-runtime
+ARG CUDA_ARCH
+ARG AMD_ARCH
+ARG CUDA_INSTALLER
+ARG DEBIAN_FRONTEND
+ARG GCC_VERSION
```
Comment (multiphaseCFD) [inline docker/Dockerfile:32]: Any reason to redefine  `CUDA_ARCH`, `AMD_ARCH`... here?

Comment (vincentmr) [reply docker/Dockerfile:32]: The defaults ARGs are defined at the top and ARGs must be imported in each stage that is not created from a previous one. Now that you mention it, I think most of them can go directly in `base-runtime`, but `PENNYLANE_VERSION` can't, for example. I'll see whether I can make it neater.

File: docker/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (vincentmr) [reply docker/Dockerfile:158]: It is not necessary.

File: docker/Dockerfile (hunk 2 above)
Comment (vincentmr) [reply docker/Dockerfile]: Works, thanks @multiphaseCFD .

**Review by multiphaseCFD — APPROVED** [review]

**Review by AmintorDusko — APPROVED** [review]
Looks good!

**Pull request merged**